<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>The PyBioMed Documentation &#8212; PyBioMed 1 documentation</title>
    
    <link rel="stylesheet" href="_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="PyBioMed 1 documentation" href="#" />
    <link rel="next" title="Overview" href="overview.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="overview.html" title="Overview"
             accesskey="N">next</a> |</li>
        <li class="nav-item nav-item-0"><a href="#">PyBioMed 1 documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="#">
              <img class="logo" src="_static/logo.png" alt="Logo"/>
            </a></p>
  <h3><a href="#">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">The PyBioMed Documentation</a><ul>
</ul>
</li>
<li><a class="reference internal" href="#indices-and-tables">Indices and tables</a></li>
</ul>

  <h4>Next topic</h4>
  <p class="topless"><a href="overview.html"
                        title="next chapter">Overview</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/index.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="the-pybiomed-documentation">
<h1>The PyBioMed Documentation<a class="headerlink" href="#the-pybiomed-documentation" title="Permalink to this headline">¶</a></h1>
<p>The python package PyBioMed is designed by <a class="reference external" href="http://home.scbdd.com/index.php?s=/Home/Index.html&amp;t=english">CBDD Group</a> (Computational Biology &amp; Drug Design Group), Xiangya School of Pharmaceutical Sciences, Central South University. To develop a powerful model for prediction tasks by machine learning algorithms such as sckit-learn, one of the most important things to consider is how to effectively represent the molecules under investigation such as small molecules, proteins, DNA and even complex interactions, by a descriptor. PyBioMed is a feature-rich package used for the characterization of various complex biological molecules and interaction samples, such as chemicals, proteins, DNA, and their interactions. PyBioMed calculates nine types of features including chemical descriptors or molecular fingerprints, structural and physicochemical features of proteins and peptides from amino acid sequence, composition and physicochemical features of DNA from their primary sequences, chemical-chemical interaction features, chemical-protein interaction features, chemical-DNA interaction features, protein-protein interaction features, protein-DNA interaction features, and DNA-DNA interaction features. We hope that the package can be used for exploring questions concerning structures, functions and interactions of various molecular data in the context of chemoinformatics, bioinformatics, and systems biology.</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="overview.html">Overview</a><ul>
<li class="toctree-l2"><a class="reference internal" href="overview.html#who-uses-pybiomed">Who uses PyBioMed?</a></li>
<li class="toctree-l2"><a class="reference internal" href="overview.html#motivation">Motivation</a></li>
<li class="toctree-l2"><a class="reference internal" href="overview.html#feature-overview">Feature overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="overview.html#the-python-programming-language">The Python programming language</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="User_guide.html">Getting Started with PyBioMed</a><ul>
<li class="toctree-l2"><a class="reference internal" href="User_guide.html#installing-the-pybiomed-package">Installing the PyBioMed package</a></li>
<li class="toctree-l2"><a class="reference internal" href="User_guide.html#getting-molecules">Getting molecules</a><ul>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#getting-molecular-structure">Getting molecular structure</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#reading-molecules">Reading molecules</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#getting-protein-sequence">Getting protein sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#reading-protein-sequence">Reading protein sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#getting-dna-sequence">Getting DNA sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#reading-dna-sequence">Reading DNA sequence</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="User_guide.html#pretreating-structure">Pretreating structure</a><ul>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#pretreating-molecules">Pretreating molecules</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#pretreating-protein-sequence">Pretreating protein sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#pretreating-dna-sequence">Pretreating DNA sequence</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="User_guide.html#calculating-molecular-descriptors">Calculating molecular descriptors</a><ul>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#calculating-descriptors">Calculating descriptors</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#calculating-molecular-fingerprints">Calculating molecular fingerprints</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="User_guide.html#calculating-protein-descriptors">Calculating protein descriptors</a><ul>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#calculating-protein-descriptors-via-functions">Calculating protein descriptors via functions</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#calculate-protein-descriptors-via-object">Calculate protein descriptors via object</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="User_guide.html#calculating-dna-descriptors">Calculating DNA descriptors</a><ul>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#calculating-dna-descriptors-via-functions">Calculating DNA descriptors via functions</a></li>
<li class="toctree-l3"><a class="reference internal" href="User_guide.html#calculating-dna-descriptors-via-object">Calculating DNA descriptors via object</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="User_guide.html#calculating-interaction-descriptors">Calculating Interaction descriptors</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="application.html">Application</a><ul>
<li class="toctree-l2"><a class="reference internal" href="application.html#application-1-prediction-of-caco-2-cell-permeability">Application 1 Prediction of Caco-2 Cell Permeability</a></li>
<li class="toctree-l2"><a class="reference internal" href="application.html#application-2-prediction-of-aqueous-solubility">Application 2 Prediction of aqueous solubility</a></li>
<li class="toctree-l2"><a class="reference internal" href="application.html#application-3-prediction-of-drugtarget-interaction-from-the-integration-of-chemical-and-protein-spaces">Application 3 Prediction of drug–target interaction from the integration of chemical and protein spaces</a></li>
<li class="toctree-l2"><a class="reference internal" href="application.html#application-4-prediction-of-protein-subcellular-location">Application 4 Prediction of protein subcellular location</a></li>
<li class="toctree-l2"><a class="reference internal" href="application.html#application-5-predicting-nucleosome-positioning-in-genomes-with-dinucleotide-based-auto-covariance">Application 5 Predicting nucleosome positioning in genomes with dinucleotide-based auto covariance</a></li>
</ul>
</li>
</ul>
</div>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules.html">PyBioMed API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="reference/PyDNA.html">PyDNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="reference/PyMolecule.html">PyMolecule</a></li>
<li class="toctree-l2"><a class="reference internal" href="reference/PyProtein.html">PyProtein</a></li>
<li class="toctree-l2"><a class="reference internal" href="reference/PyPretreat.html">PyPretreat</a></li>
<li class="toctree-l2"><a class="reference internal" href="reference/PyInteraction.html">PyInteraction</a></li>
<li class="toctree-l2"><a class="reference internal" href="reference/PyGetMol.html">PyGetMol</a></li>
<li class="toctree-l2"><a class="reference internal" href="reference/test.html">test module</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="test.html">Testing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="test.html#requirements-for-testing">Requirements for testing</a></li>
<li class="toctree-l2"><a class="reference internal" href="test.html#testing-an-installed-package">Testing an installed package</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="download.html">Download</a><ul>
<li class="toctree-l2"><a class="reference internal" href="download.html#python-package">Python Package</a></li>
<li class="toctree-l2"><a class="reference internal" href="download.html#documentation">Documentation</a></li>
<li class="toctree-l2"><a class="reference internal" href="download.html#the-introduction-of-descriptors">The introduction of descriptors</a></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></li>
<li><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></li>
<li><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></li>
</ul>
<div class="figure align-center">
<a class="reference internal image-reference" href="_images/logocbdd.png"><img alt="_images/logocbdd.png" src="_images/logocbdd.png" style="width: 500px;" /></a>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="overview.html" title="Overview"
             >next</a> |</li>
        <li class="nav-item nav-item-0"><a href="#">PyBioMed 1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright Copyright @ 2015-2016 Computational Biology &amp; Drug Design Group.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.9.
    </div>
  </body>
</html>